src/app/shared/services/incentive-leaders.service.ts

```diff
@@ -0,0 +1,278 @@
+  providedIn: 'root'
+})
+export class IncentiveLeadersService {
+  private static readonly STORAGE_KEY = 'IncentiveLeadersService';
+  private static readonly UPDATE_INTERVAL = 60 * 60 * 1000;
+
+  etherscanUrl = environment.etherscanUrl;
+
+  private esService: ElasticSearchService<any>;
+  private sellersWithMostPopularFilesSubject = new BehaviorSubject<any[]>(undefined);
+  private sellersWithHighestAverageRatingSubject = new BehaviorSubject<any[]>(undefined);
+  private sellersWithMostPopularFileTodaySubject = new BehaviorSubject<any[]>(undefined);
+  private sellersWithMostPopularFileYesterdaySubject = new BehaviorSubject<any[]>(undefined);
+
+  constructor(
+    private readonly http: HttpClient,
+    private readonly dataProductListService: DataProductListService,
+    private readonly storageService: StorageService) {
+    this.esService = new ElasticSearchService<any>(this.http, null);
+
+    this.initialize();
+  }
+
+  getSellersWithMostPopularFiles(): Observable<any[]> {
+    return this.sellersWithMostPopularFilesSubject.asObservable();
+  }
+
+  getSellersWithHighestAverageRating(): Observable<any[]> {
+    return this.sellersWithHighestAverageRatingSubject.asObservable();
+  }
+
+  getSellersWithMostPopularFileTodaySubject(): Observable<any[]> {
+    return this.sellersWithMostPopularFileTodaySubject.asObservable();
+  }
+
+  getSellersWithMostPopularFileYesterdaySubject(): Observable<any[]> {
+    return this.sellersWithMostPopularFileYesterdaySubject.asObservable();
+  }
+
+  private fetchFilePipe(groupName: string): OperatorFunction<any, {}[]> {
+    return flatMap((result: any) => {
+      const buckets = result.aggregations[ groupName ].buckets;
+
+      if (!buckets.length) {
+        return of([]);
+      }
+
+      return forkJoin(buckets.map(bucket => {
+        return this.dataProductListService.getDataProduct(bucket.key)
+          .pipe(map(product => {
+            bucket.product = {
+              ownerAddress: product.ownerAddress,
+              address: product.address,
+              finalisedOrdersNumber: product.orders.filter(order => order.finalised).length
+            };
+            return bucket;
+          }));
+      }));
+    });
+  }
+
+  private initialize() {
+    const config = this.readFromStore();
+
+    if (!config.updateTimestamp || this.getReloadTimeout(config.updateTimestamp) < 0) {
+      this.fetchData();
+
+      return;
+    }
+
+    this.sellersWithMostPopularFilesSubject.next(config.sellersWithMostPopularFiles);
+    this.sellersWithHighestAverageRatingSubject.next(config.sellersWithHighestAverageRating);
+    this.sellersWithMostPopularFileTodaySubject.next(config.sellersWithMostPopularFileToday);
+    this.sellersWithMostPopularFileYesterdaySubject.next(config.sellersWithMostPopularFileYesterday);
+
+    this.setTimeout(config.updateTimestamp);
+  }
+
+  private async fetchData(): Promise<void> {
+    this.esService.searchRaw('user', topRatedSellerQuery)
+      .subscribe(result => {
```

Comment:
I would reconsider go get rid off these subscriptions and subjects and just return observables with `do` operator to update storage.